make/data/asan/asan_default_options.c

```diff
@@ -52,6 +52,12 @@ ATTRIBUTE_DEFAULT_VISIBILITY ATTRIBUTE_USED const char* __asan_default_options()
   return
 #ifndef LEAK_SANITIZER
```

Comment:
I'm trying to understand this part. The test for `LEAK_SANITIZER` was put in place in this file, before it was ever generated by the build system..? Was this in anticipation for this PR, or intended for the user to add an extra `-DLEAK_SANITIZER` to the compiler flags manually?

Also, this patch changes the values for asan, regardless of the state of lsan. Is this required to get the lsan patch working, or is this additional, separate fixes to asan being "smuggled" into this PR? (I don't mind a certain amount of "smuggling", since sometimes it is too tedious to open a separate JBS issue for minor, separate fixes one found out during development), but it is nice to be explicit about it.

And finally, now that the if statement also has an else, maybe reverse it to an `#ifdef` instead of `#ifndef`..?

Apart from this, the build changes look good.